test/unit_test/engine/test_footprint.py

```diff
@@ -45,6 +46,14 @@ def test_pareto_frontier(self):
         assert len(pareto_frontier_fp.nodes) == 2
         assert all(v.is_pareto_frontier for v in pareto_frontier_fp.nodes.values())
 
+    def test_empty_pareto_frontier(self):
+        # set all `is_goals_met` as false
+        unmet_goals_nodes = deepcopy(self.fp.nodes)
+        for v in unmet_goals_nodes.values():
+            v.metrics.if_goals_met = False
+        new_fp = Footprint(nodes=unmet_goals_nodes)
+        assert not new_fp.create_pareto_frontier()
```

Comment:
>not [](http://example.com/codeflow?start=15&length=3)

assert is None instead of empty dict?

Reply:
`create_pareto_fromtier` return `Footprint` not `FootprintNode`. `Footprint` is a class where I think `not new_fp` is equal with `new_fp is None`